yarn-project/archiver/src/archiver/l1_to_l2_message_store.ts

```diff
@@ -61,7 +61,7 @@ export class PendingL1ToL2MessageStore extends L1ToL2MessageStore {
     const messageKeyBigInt = messageKey.value;
     const msgAndCount = this.store.get(messageKeyBigInt);
     if (!msgAndCount) {
-      return;
+      throw new Error(`Message with key ${messageKeyBigInt} not found in store`);
```

Comment:
Have you tried executing blocks that are not full of messages? I'm expecting that you will run into the case you are testing in `handles case when removing a message that doesn't exist` from 0 messages.